openedx/features/pakx/lms/pakx_admin_app/views.py

```diff
@@ -419,3 +421,30 @@ def get(self, *args, **kwargs):  # pylint: disable=unused-argument
             user_info['role'] = TRAINING_MANAGER if user_groups.name == GROUP_TRAINING_MANAGERS else ORG_ADMIN
 
         return Response(status=status.HTTP_200_OK, data=user_info)
+
+
+class CourseListAPI(generics.ListAPIView):
+    authentication_classes = [SessionAuthentication]
+    permission_classes = [CanAccessPakXAdminPanel]
+    pagination_class = PakxAdminAppPagination
+    serializer_class = CoursesSerializer
+
+    PakxAdminAppPagination.page_size = 3
+
+    instructors = {}
+
+    def get_serializer_context(self):
+        context = super(CourseListAPI, self).get_serializer_context()
+        context.update({"instructors": self.instructors})
+        return context
+
+    def get_queryset(self):
+        queryset = CourseOverview.objects.all()
+        course_access_role_qs = CourseAccessRole.objects.filter(course_id__in=queryset.values_list('id'))
```

Comment:
add 'user' & 'user__profile' in select_related to avoid queries in loop. 
Kindly check this in Django Silk before making the PR whenever you make such changes

Reply:
select_related still missing ?